data_analysis/plots.py

```diff
@@ -611,55 +622,184 @@ def make_ratio_map(
         )
 
 
-def run_mvp() -> None:
-    """Reproduce maps and JSONs from MVP launch on 2022-11-11.
+def plot_and_save(
+    summary_gdf_geo: gpd.GeoDataFrame,
+    summary_kwargs: dict,
+        save_name: str) -> None:
+    """Plot a map from the data input and save it
+
+    Args:
+        summary_gdf_geo (gpd.GeoDataFrame): A GeoDataFrame of the summary data
+            from compare_scheduled_and_rt.main or from the saved csv of the
+            MVP data in the data_output directory
+        summary_kwargs (dict): A dictionary of keyword arguments passed to
+            geopandas.GeoDataFrame.explore method
+        save_name (str): The name of the saved map
     """
-    logger.info("Creating GeoDataFrame")
-    gdf = static_gtfs_analysis.main()
+    save_name = f"{save_name}_{summary_kwargs['column']}"
+    make_map(
+        summary_gdf_geo=summary_gdf_geo,
+        summary_kwargs=summary_kwargs,
+        save_name=save_name
+    )
 
-    logger.info(
-            "Loading data for MVP."
-            " Date range is between 2022-05-20 and 2022-11-05"
-        )
-    combined_long_df = pd.read_csv(
-        DATA_PATH / 'combined_long_df_2022-11-06.csv'
+    path_name = create_save_path(save_name, DATA_PATH)
+    # Take only the columns related to summary_kwargs['column']
+    # and those used in the map
+    first_cols = summary_gdf_geo.columns[:2].tolist()
+    last_cols = summary_gdf_geo.columns[-10:].to_list()
+    kwargs_cols = summary_gdf_geo.columns[
+        summary_gdf_geo.columns.str.startswith(summary_kwargs['column'])
+    ].tolist()
+    cols = first_cols + kwargs_cols + last_cols
+    summary_gdf_geo[cols].to_file(f'{path_name}.json', driver='GeoJSON')
+    summary_gdf_geo[cols].to_html(f'{path_name}_table.html', index=False)
+
+
+def calculate_trips_per_rider(
```

Comment:
I think long term it might be good to break actual calculations out from this and have a separate place for those? (Like, ideally `plots.py` is just for mechanics of generating plots.) Not sure that needs to block merging this right now, but `plots.py` is just getting pretty massive.

Reply:
Agreed that `plots.py` is getting too big. I can move the calculations into a separate file for now and create a class to avoid duplicate function arguments.